tests/cephfs/cephfs_nfs/test_cephfs_nfs_ha_snapshot_lifecycle.py

```diff
@@ -0,0 +1,342 @@
+            )
+            export_path = f"{subvol_path}"
+            nfs_export_name = "/export_" + "".join(
+                secrets.choice(string.digits) for i in range(3)
+            )
+            log.info(f"Export create for subvolume {subvol_name}")
+            client1.exec_command(
+                sudo=True,
+                cmd=f"ceph nfs export create cephfs {nfs_name} "
+                f"{nfs_export_name} {fs_name} path={export_path}",
+            )
+            export_name_list.append(nfs_export_name)
+            out, rc = client1.exec_command(
+                sudo=True, cmd=f"ceph nfs export get {nfs_name} {nfs_export_name}"
+            )
+            json.loads(out)
+
+            if nfs_export_name not in out:
+                raise CommandFailed("Failed to create nfs export")
+
+        log.info("Perform NFS mount of subvolumes")
+        for nfs_export_name in export_name_list:
+            nfs_mounting_dir = "/mnt/nfs_" + "".join(
+                secrets.choice(string.ascii_uppercase + string.digits) for i in range(3)
+            )
+            client1.exec_command(sudo=True, cmd=f"mkdir -p {nfs_mounting_dir}")
+            command = f"mount -t nfs -o port=2049 {virtual_ip}:{nfs_export_name} {nfs_mounting_dir}"
+            client1.exec_command(sudo=True, cmd=command, check_ec=False)
+            nfs_mount_list.append(nfs_mounting_dir)
+
+        log.info("Run IOs on both subvolumes")
+        dir_name = "dd_io_dir"
+        for mount in nfs_mount_list:
+            client1.exec_command(sudo=True, cmd=f"mkdir -p {mount}/{dir_name}")
+
+        log.info("Create snapshots on subvolumes at regular intervals during IO")
+        snap_list = [f"nfs_snap_{i}" for i in range(5)]
+        with parallel() as p:
+            p.spawn(write_io_wrapper, client1, nfs_mount_list, dir_name, 2)
+            p.spawn(create_snapshot_wrapper, client1, fs_util, subvol_list, snap_list)
+
+        log.info("Verify all snapshots are created")
+        for subvol in subvol_list:
+            for snap_name in snap_list:
+                snap_info_cmd = (
+                    f"ceph fs subvolume snapshot info {fs_name} {subvol} {snap_name}"
+                )
+                snap_info_cmd = f"{snap_info_cmd} --group_name subvolume_group1"
+                cmd_out, cmd_rc = client1.exec_command(
+                    sudo=True, cmd=snap_info_cmd, check_ec=True
+                )
+                log.info(cmd_out)
+
+        log.info("Verify all 5 snapshots are listed in mount points on both volumes")
+        validate_snap_nfs(client1, nfs_mount_list, snap_list)
+
+        log.info("Run Read IO from snapshot dir on both subvolumes")
+
+        for mount in nfs_mount_list:
+            snap_name = get_snap_name(client1, f"{mount}/.snap", snap_list[-1])
+            io_path = f"{mount}/.snap/{snap_name}/{dir_name}"
+            read_io(client1, io_path)
+
+        log.info("Run HA failover by rebooting active nfs node")
```

Comment:
NFS HA failover cannot be validated by rebooting the node that has a virtual IP,  NFS Nodes will be distributed for load balancing. For each directory path(or a subvolume in this scenario) - one of the NFS nodes will be active. So, it's better we run IO's on all the exports in parallel and perform a reboot of all the NFS nodes one after the other and validate the mount points are intact until all the NFS nodes are rebooted and back to original state.